src/js/flagset.ts

```diff
@@ -802,7 +802,7 @@ export class FlagSet {
     const out = [];
     for (const [key, section] of sections.sortedEntries()) {
       let sec = key;
-      for (const subkey of ['', '?']) {
+      for (const subkey of ['', '?', '!']) {
```

Comment:
This should work for now, but I intend to add other modes in the future.  I think the proper fix is to _mostly_ revert the change to

```
for (const [subkey, subsection] of section.sortedEntries()) {
  sec += subkey + subsection.sort().join('');
}
```

Basically, the thing that was missing before was that I wasn't sorting the entries by key, so '?' would appear before '' which breaks parsing.  Since the default iteration order is determined by the order in which the entries were added, this can show up in some perverse cases.  Further, this is supposed to be a _canonical_ string, so nondeterministic ordering is no good.  The correct fix is to make sure we do the sorted iteration.

I assume you made this patch with the inline edit button in the github UI, so I'm not sure how easy it is for you to update it in place.  So I'll go ahead and fix it myself.  Thanks a ton for finding the issue!

Reply:
Yeah, I just used the UI editor.  Thanks for the explanation, too!  Glad it's cleaned up now - FYI I tested it on live now and I can make the seeds I want to again and also can't replicate the bug Axe was trying to fix (but I'm not 100% sure I'm doing it the same way he ran into it).